src/llmcompressor/pytorch/model_load/helpers.py

```diff
@@ -106,6 +107,12 @@ def save_model_and_recipe(
     :param save_safetensors: whether to save as safetensors or pickle (bin)
     :param save_compressed: whether to compress sparse weights on disk
     """
+    if is_model_quantized(model):
+        from llmcompressor.transformers.sparsification.compressed_tensors_utils import (
+            modify_save_pretrained,
+        )
+
+        modify_save_pretrained(model)
```

Comment:
We should probably also take this opportunity to clean-up some of what happens in the `save_pretrained_wrapper` (e.g. seem to always calculate sparsity statistics by default)

Also, oneshot gives the option to provide an output dir, which I think also saves the model:
https://github.com/vllm-project/llm-compressor/blob/b3c474f89621bed5fc2d779226e50cb1b34dd50b/tests/llmcompressor/transformers/obcq/test_consecutive_runs.py#L30 but not 100% sure.